3D/senior-csharp-test/Assets/Cache/CacheEntry.cs

```diff
@@ -10,6 +10,18 @@ public class CacheEntry<T>
 
 	public CacheEntry( string id, string version, T data, string lastUseDate )
 	{
+		if( string.IsNullOrEmpty( id ) )
+			throw new NullReferenceException( nameof( id ) );
+
+		if( string.IsNullOrEmpty( version ) )
+			throw new NullReferenceException( nameof( version ) );
+
+		if( string.IsNullOrEmpty( lastUseDate ) )
+			throw new NullReferenceException( nameof( lastUseDate ) );
+
+		if( data.Equals( default( T ) ) )
```

Comment:
`data` should be checked for null before using

Reply:
Modified based on https://stackoverflow.com/questions/65351/null-or-default-comparison-of-generic-argument-in-c-sharp/864860#864860